core/src/main/java/org/apache/carbondata/core/indexstore/DataMapStoreManager.java

```
@@ -69,20 +70,20 @@ public TableDataMap getDataMap(AbsoluteTableIdentifier identifier, String dataMa
    * @param mapType
    * @return
    */
-  public TableDataMap createTableDataMap(AbsoluteTableIdentifier identifier, DataMapType mapType,
-      String dataMapName) {
-    Map<String, TableDataMap> map = dataMapMappping.get(mapType);
+  public AbstractTableDataMap createTableDataMap(AbsoluteTableIdentifier identifier,
+      DataMapType mapType, String dataMapName) {
+    Map<String, AbstractTableDataMap> map = dataMapMappping.get(mapType);
     if (map == null) {
       map = new HashMap<>();
       dataMapMappping.put(mapType, map);
     }
-    TableDataMap dataMap = map.get(dataMapName);
+    AbstractTableDataMap dataMap = map.get(dataMapName);
     if (dataMap != null) {
       throw new RuntimeException("Already datamap exists in that path with type " + mapType);
     }
 
     try {
-      //TODO create datamap using @mapType.getClassName())
+      dataMap = (AbstractTableDataMap) (Class.forName(mapType.getClassName()).newInstance());
     } catch (Exception e) {
       LOGGER.error(e);
```

should not ignore the exception